src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,190 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class AuditConfig {
+    private static final List<String> DEFAULT_IGNORED_USERS = Collections.singletonList("kibanaserver");
+    private static final List<String> DEFAULT_DISABLED_CATEGORIES =
+            Arrays.asList(AuditCategory.AUTHENTICATED.toString(),
+                    AuditCategory.GRANTED_PRIVILEGES.toString());
+
+    private final boolean isRestApiAuditEnabled;
+    private final boolean isTransportAuditEnabled;
+    private final boolean resolveBulkRequests;
+    private final boolean logRequestBody;
+    private final boolean resolveIndices;
+    private final boolean excludeSensitiveHeaders;
+    private final Set<String> ignoredAuditUsers;
+    private final Set<String> ignoredComplianceUsersForRead;
+    private final Set<String> ignoredComplianceUsersForWrite;
+    private final Set<String> ignoreAuditRequests;
+    private final EnumSet<AuditCategory> disabledRestCategories;
+    private final EnumSet<AuditCategory> disabledTransportCategories;
+    private final String opendistrosecurityIndex;
+
+    private AuditConfig(final boolean isRestApiAuditEnabled,
+                        final boolean isTransportAuditEnabled,
+                        final boolean resolveBulkRequests,
+                        final boolean logRequestBody,
+                        final boolean resolveIndices,
+                        final boolean excludeSensitiveHeaders,
+                        final Set<String> ignoredAuditUsers,
```

Comment:
I feel whoever is calling the constructor should deal with that. In this case, getConfig(settings). What if "NONE" is one of the users? It would be wrong to handle this in constructor. What are your thoughts?

Reply:
OK, let's assume that "NONE" is a valid input in the constructor and it means an empty list only in settings.